### ISSUE TITLE
File encoding and line endings

### ISSUE THREAD

**Issue by whatsondoc** [Azure/azure-batch-cli-extensions#51]
Hello,

I've noticed on a couple of different occasions when using --template option on the CLI an error would be returned showing:

benji$ az batch job create --template job-template.json --account-name BATCH_NAME --account-endpoint BATCH_ENDPOINT
The behavior of this command has been altered by the following extension: azure_batch_cli_extensions
You are using an experimental feature {Job Template}.
Invalid JSON file: No JSON object could be decoded

Having investigated this it appears a couple of causes are in play:

- UNIX line endings
- File encoding

Can we have the service better interpret/handle template files? 

json files that have encountered this error:
:: https://github.com/whatsondoc/azure-batch/tree/master/pool/marketplace-image (pool and job templates)
:: https://github.com/karlpodesta/azure-batch-hpc-lab (pool and job templates)

**Comment by xingwu1** [comment]
This issue should be fixed at latest version cli (2.0.31) and batch-extension (2.2.1). Please upgrade to latest cli and extension.

However, there has a compatible issue between latest cli and batch-extensions, so you have to manually update dependency `azure-batch-extensions` to 1.1.1.

If you fresh install azure-cli in Ubuntu, you can run following command to update the component:
```
sudo /opt/az/bin/python3 -m pip install -U azure-batch-extensions==1.1.1 
```
If you run azure-cli on windows, the following command is used to update the component:
```
"C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe" -m pip install -U azure-batch-extensions==1.1.1 
```
The compatible issue should be fixed at next azure-cli release.

**Comment by xingwu1** [comment]
This issue has been address with latest azure-cli and batch extensions.
